src/accounts/nuls2.ts

```diff
@@ -132,10 +139,93 @@ export async function ImportAccountFromPrivateKey(
     privateKey: string,
     { chain_id = 1, prefix = "NULS" }: NULS2ImportConfig = { chain_id: 1, prefix: "NULS" },
 ): Promise<NULS2Account> {
-    const pub = privateKeyToPublicKey(Buffer.from(privateKey, "hex"));
+    const pub = secp256k1.publicKeyCreate(Buffer.from(privateKey, "hex"));
     const publicKey = Buffer.from(pub).toString("hex");
 
     const hash = publicKeyToHash(pub, { chain_id: chain_id });
     const address = addressFromHash(hash, prefix);
     return new NULS2Account(address, publicKey, privateKey);
 }
+
+/********************************
+ * NULS2 TOOLS METHODS
+ ********************************/
+
+/**
+ * Extract an address from a given hash.
+ *
+ * @param hash The hash containing the address.
+ * @param prefix The optional address prefix.
+ */
+export function addressFromHash(hash: Uint8Array, prefix?: string): string {
+    const address = bs58.encode(Buffer.concat([hash, Buffer.from([getXOR(hash)])]));
+
+    if (prefix) return prefix + String.fromCharCode(prefix.length + 96) + address;
+    return address;
+}
+
+/**
+ * Creates a XOR of an array.
+ *
+ * @param body The array to XOR.
+ */
+export function getXOR(body: Uint8Array): number {
+    return body.reduce((xor, i) => (xor ^= i), 0);
+}
+
+/**
+ * Creates a hash from a message.
+ *
+ * @param message The message used to create the hash.
+ * @param messagePrefix The optional message's hash prefix.
+ */
+export function magicHash(message: Buffer, messagePrefix?: string | Buffer): Buffer {
+    if (!messagePrefix) messagePrefix = "\u0018NULS Signed Message:\n";
+    if (!Buffer.isBuffer(messagePrefix)) messagePrefix = Buffer.from(messagePrefix);
+
```

Comment:
Can you use [default paramaters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) instead ? It is more explicit.